### PR TITLE
[GSK-2465] Separate args for leaderboard and inference token

### PR DESCRIPTION
**Pull request by cy-moi**
Features:
- Separate args to set leaderboard dataset id, if none, do not commit results.
- Add inference token arg in config file, when inference_type is inference api this separate token has to be provide otherwise will raise error.

### CODE REVIEW

File: giskard_cicd/cli.py [hunk 1]
```diff
@@ -126,9 +131,9 @@ def main():
     else:
         rendered_report = report.to_html()
 
-    if args.output_portal == "huggingface":
+    if args.output_portal == "huggingface" and args.leaderbord_dataset:
```
Comment (Inokinoki) [inline giskard_cicd/cli.py:134]: I think the discussion could be published without `leaderboard_dataset`.

```suggestion
    if args.output_portal == "huggingface":
        # Create the discussion
        # ...
        if args.leaderboard_dataset:
            # Init dataset commit scheduler
            # ...
```

**Review by Inokinoki — CHANGES_REQUESTED** [review]

File: giskard_cicd/cli.py (hunk 1 above)
Comment (cy-moi) [reply giskard_cicd/cli.py]: Fixed

File: giskard_cicd/cli.py [hunk 2]
```diff
@@ -63,6 +63,10 @@ def main():
         "--persistent_scan", help="Persistent scan report.", type=bool, default=False
     )
 
+    parser.add_argument(
+        "--leaderbord_dataset", help="The leaderboard dataset to push the report to."
```
Comment (Inokinoki) [inline giskard_cicd/cli.py:67]: typo -> leaderboard

File: giskard_cicd/cli.py [hunk 3]
```diff
@@ -143,15 +144,25 @@ def main():
...
-            scheduler,
-            args.model,
-            args.dataset,
-            args.dataset_config,
-            args.dataset_split,
-            discussion,
-            report,
-        )
+
+        if args.leaderbord_dataset:  # Commit to leaderboard dataset
```
Comment (Inokinoki) [inline giskard_cicd/cli.py:148]: typo

File: giskard_cicd/cli.py [hunk 4]
```diff
@@ -143,15 +144,25 @@ def main():
...
+            scheduler = init_dataset_commit_scheduler(
+                hf_token=args.hf_token, dataset_id=args.leaderbord_dataset
```
Comment (Inokinoki) [inline giskard_cicd/cli.py:150]: same

**Review by Inokinoki — CHANGES_REQUESTED** [review]
LGTM

Just a typo and resulting typos in arg name

**Review by Inokinoki — APPROVED** [review]

**Pull request merged**